### PR TITLE
test: some tests for material, common, core and bucket module

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: stacks/spica/packages/core/layout/error.interceptor.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {Router} from "@angular/router";
 import {Observable} from "rxjs";
 import {tap} from "rxjs/operators";
 
+// @Todo Test
```
Comment (tthiscafer) [inline stacks/spica/packages/core/layout/error.interceptor.ts]: will the test be written here?

File: stacks/spica/packages/core/layout/error/error.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    queryParams: of({
+      status: "Test Status Number",
+      statusText: "Test Status Text",
+      message: "Test Message"
+    })
+  };
+
+  let location: Location;
+  let fixture: ComponentFixture<ErrorPageComponent>;
+  // let component : ErrorPageComponent
```
Comment (tthiscafer) [inline stacks/spica/packages/core/layout/error/error.page.spec.ts]: comment lines must be removed

File: stacks/spica/packages/core/layout/error/error.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      providers: [
+        {
+          provide: ActivatedRoute,
+          useValue: queryParams
+        }
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ErrorPageComponent);
+    // component = fixture.componentInstance;
```
Comment (tthiscafer) [inline stacks/spica/packages/core/layout/error/error.page.spec.ts]: comment lines must be removed

File: stacks/spica/packages/core/layout/error/error.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import {TestBed, ComponentFixture, tick, fakeAsync} from "@angular/core/testing";
+import {MatButtonModule} from "@angular/material/button";
+import {ErrorPageComponent} from "./error.page";
+import {ActivatedRoute} from "@angular/router";
+import {of} from "rxjs";
+import {RouterTestingModule} from "@angular/router/testing";
+import {Location} from "@angular/common";
+
+//DONE
```
Comment (tthiscafer) [inline stacks/spica/packages/core/layout/error/error.page.spec.ts]: unnecessary comment lines should be removed

**Comment by thesayyn** [comment]
@htuna07  Take an overall look

**Comment by thesayyn** [comment]
ping @tthiscafer

**Comment by thesayyn** [comment]
cc @htuna07 tests are failing

**Comment by thesayyn** [comment]
@htuna07 Some tests are failing

File: stacks/spica/packages/common/input/components/boolean/boolean.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      })
+      .compileComponents();
+  });
+
+  describe("basic behavior", () => {
+    let fixture: ComponentFixture<BooleanComponent>;
+    let labelElement: HTMLLabelElement;
+    let slideToggle: MatSlideToggle;
+
+    beforeEach(fakeAsync(() => {
```
Comment (htuna07) [inline stacks/spica/packages/common/input/components/boolean/boolean.component.spec.ts]: Is **fakeAsync** necessary?

File: stacks/spica/packages/common/input/components/boolean/boolean.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      fixture = TestBed.createComponent(BooleanComponent);
+      fixture.detectChanges();
+      labelElement = fixture.debugElement.query(By.css("label")).nativeElement;
+      slideToggle = fixture.debugElement.query(By.css("mat-slide-toggle")).componentInstance;
+    }));
+
+    it("should default to default value if undefined", () => {
+      const fixture = TestBed.createComponent(BooleanComponent);
```
Comment (htuna07) [inline stacks/spica/packages/common/input/components/boolean/boolean.component.spec.ts]: Fixture has been created on beforeEach block already.

File: stacks/spica/packages/common/input/components/date/date.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      const formFieldElem = fixture.debugElement.query(By.css("mat-form-field")).nativeElement;
+      expect(formFieldElem.classList).toContain("ng-untouched");
+      expect(formFieldElem.classList).toContain("ng-pristine");
+      expect(formFieldElem.classList).toContain("ng-valid");
+    });
+
+    it("should open picker on click", () => {
+      expect(document.body.querySelector("owl-date-time-container")).toBeNull();
+      const input = fixture.debugElement.query(By.css("input"));
+      input.nativeElement.click();
```
Comment (htuna07) [inline stacks/spica/packages/common/input/components/date/date.component.spec.ts]: Same process could be done with one line code.

File: stacks/spica/packages/common/input/components/number/number.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+      })
+      .compileComponents();
+    fixture = TestBed.createComponent(NumberComponent);
+    fixture.detectChanges();
+  });
+
+  describe("basic behavior", () => {
+    it("should create and set the model name", () => {
+      const input = fixture.debugElement.query(By.directive(NgModel)).injector.get(NgModel);
+      expect(input.path).toEqual(["test"]);
```
Comment (htuna07) [inline stacks/spica/packages/common/input/components/number/number.component.spec.ts]: Same process could be done with one line code.

File: stacks/spica/src/passport/pages/identify/identify.component.spec.ts [hunk 9]
```diff
@@ -1,22 +1,157 @@
...
+        .nativeElement.click();
+
+      tick();
+      fixture.detectChanges();
+
+      expect(routerSpy).toHaveBeenCalledTimes(1);
+      expect(routerSpy).toHaveBeenCalledWith(["/dashboard"]);
+    }));
+
+    it("should identify successfully when clicked login button after filled form", fakeAsync(async () => {
```
Comment (htuna07) [inline stacks/spica/src/passport/pages/identify/identify.component.spec.ts]: fakeAsync is useless

File: stacks/spica/src/storage/pages/index/index.component.ts [hunk 10]
```diff
@@ -77,15 +77,17 @@ export class IndexComponent implements OnInit {
...
-        }
+        () => this.uploadDone()
       );
     }
   }
 
+  uploadDone() {
+    this.sortStorage({direction: "desc", name: "_id"});
+    this.progress = undefined;
+    this.refresh.next();
```
Comment (htuna07) [inline stacks/spica/src/storage/pages/index/index.component.ts]: next() has been called from sortStorage already.

**Comment by thesayyn** [comment]
cc @htuna07 Tests are still failing. Please refer to errors in CI.

**Comment by thesayyn** [comment]
Note: Some tests were disabled due to insufficient CI resources.

**Pull request merged**